Index: src/App.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/App.js b/src/App.js
--- a/src/App.js	(revision c879766a2f8f56cb284b0207a693aa66818df1b2)
+++ b/src/App.js	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -7,17 +7,22 @@
 
     const[movies,setMovies] = useState(getMovies())
 
-    const onDeleteHandler =(movie) =>{
+    const onDeleteHandler =(movie) => setMovies(movies.filter(m => m._id !== movie._id))
+    const onLikeHandler = (movie) =>{
+    let updatedMovies = movies.map(m => {
+        if(m._id===movie._id) {
+            return {...m,liked:!m.liked} // this makes the opposite of the movie unliked
+        }
+        return m;
+     })
 
-        setMovies(movies.filter(m=>m._id!==movie._id))
-
+        setMovies(updatedMovies)
     }
 
     return (
 
         <main className="container">
-            {console.log(movies)}
-           <Movie movies={movies}  onDeleteHandler={onDeleteHandler}/>
+           <Movie movies={movies}  onDeleteHandler={onDeleteHandler} onLikeHandler={onLikeHandler} />
 
         </main>
     );
Index: src/Starter Code/services/fakeMovieService.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Starter Code/services/fakeMovieService.js b/src/Starter Code/services/fakeMovieService.js
--- a/src/Starter Code/services/fakeMovieService.js	(revision c879766a2f8f56cb284b0207a693aa66818df1b2)
+++ b/src/Starter Code/services/fakeMovieService.js	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -7,63 +7,72 @@
     genre: { _id: "5b21ca3eeb7f6fbccd471818", name: "Action" },
     numberInStock: 6,
     dailyRentalRate: 2.5,
-    publishDate: "2018-01-03T19:04:28.809Z"
+    publishDate: "2018-01-03T19:04:28.809Z",
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd471816",
     title: "Die Hard",
     genre: { _id: "5b21ca3eeb7f6fbccd471818", name: "Action" },
     numberInStock: 5,
-    dailyRentalRate: 2.5
+    dailyRentalRate: 2.5,
+    liked:true
   },
   {
     _id: "5b21ca3eeb7f6fbccd471817",
     title: "Get Out",
     genre: { _id: "5b21ca3eeb7f6fbccd471820", name: "Thriller" },
     numberInStock: 8,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd471819",
     title: "Trip to Italy",
     genre: { _id: "5b21ca3eeb7f6fbccd471814", name: "Comedy" },
     numberInStock: 7,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd47181a",
     title: "Airplane",
     genre: { _id: "5b21ca3eeb7f6fbccd471814", name: "Comedy" },
     numberInStock: 7,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd47181b",
     title: "Wedding Crashers",
     genre: { _id: "5b21ca3eeb7f6fbccd471814", name: "Comedy" },
     numberInStock: 7,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd47181e",
     title: "Gone Girl",
     genre: { _id: "5b21ca3eeb7f6fbccd471820", name: "Thriller" },
     numberInStock: 7,
-    dailyRentalRate: 4.5
+    dailyRentalRate: 4.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd47181f",
     title: "The Sixth Sense",
     genre: { _id: "5b21ca3eeb7f6fbccd471820", name: "Thriller" },
     numberInStock: 4,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   },
   {
     _id: "5b21ca3eeb7f6fbccd471821",
     title: "The Avengers",
     genre: { _id: "5b21ca3eeb7f6fbccd471818", name: "Action" },
     numberInStock: 7,
-    dailyRentalRate: 3.5
+    dailyRentalRate: 3.5,
+    liked:false
   }
 ];
 
Index: src/components/Movie/Movie.Component.jsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Movie/Movie.Component.jsx b/src/components/Movie/Movie.Component.jsx
--- a/src/components/Movie/Movie.Component.jsx	(revision c879766a2f8f56cb284b0207a693aa66818df1b2)
+++ b/src/components/Movie/Movie.Component.jsx	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -7,6 +7,10 @@
      props.onDeleteHandler(movie)
     }
 
+    const onLike = (movie) =>{
+       props.onLikeHandler(movie)
+    }
+
     return (
         <div>
 
@@ -19,6 +23,7 @@
                     <th scope="col">Stock</th>
                     <th scope="col">Rate</th>
                     <th scope="col"/>
+                    <th scope="col"/>
                 </tr>
                 </thead>
                 <tbody>
@@ -27,6 +32,7 @@
                     <MovieTitle key={movie._id}
                                 movie={movie}
                                 onDelete = {onDelete}
+                                onLike={onLike}
 
                     />
                 )}
Index: src/components/Movie/MovieTitle.Component.jsx
===================================================================
diff --git a/src/components/Movie/MovieTitle.Component.jsx b/src/components/Movie/MovieTitle.Component.jsx
--- a/src/components/Movie/MovieTitle.Component.jsx	(revision c879766a2f8f56cb284b0207a693aa66818df1b2)
+++ b/src/components/Movie/MovieTitle.Component.jsx	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -1,11 +1,16 @@
 import React from 'react';
 import Button from "../UI/Button.UI.Jsx";
+import Like from "../common/Like.Component";
 
 const MovieTitle = (props) => {
-    const {genre, _id, numberInStock, title, dailyRentalRate} = props.movie;
+    const {genre, _id, numberInStock, title, dailyRentalRate,liked} = props.movie;
     const onDelete = ()=>{
                props.onDelete(props.movie)
         }
+
+    const onLike = () =>{
+       props.onLike(props.movie)
+    }
         return (
             <tr className='table-info'>
 
@@ -13,6 +18,7 @@
                     <td>{genre.name}</td>
                     <td>{numberInStock}</td>
                     <td>{dailyRentalRate}</td>
+                    <td><Like liked={liked} onLike={onLike}/></td>
                     <Button onDelete={onDelete} movie={props.movie} buttonType='btn btn-danger'>Delete</Button>
 
 
Index: src/components/common/Like.Component.jsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/common/Like.Component.jsx b/src/components/common/Like.Component.jsx
new file mode 100644
--- /dev/null	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
+++ b/src/components/common/Like.Component.jsx	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -0,0 +1,14 @@
+import React from 'react';
+// input whether its liked or not.
+const Like = (props) => {
+    let classes='fa fa-heart'
+    const onLikeHandler=() =>{
+      props.onLike()
+    }
+    if (!props.liked) classes+='-o';
+    return (
+        <i className={classes} aria-hidden='true' onClick={onLikeHandler}/>
+    );
+};
+
+export default Like;
\ No newline at end of file
Index: src/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/index.js b/src/index.js
--- a/src/index.js	(revision c879766a2f8f56cb284b0207a693aa66818df1b2)
+++ b/src/index.js	(revision 5ef158c665986323884d8b5062e12b8af808f79f)
@@ -4,6 +4,7 @@
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 import 'bootswatch/dist/lumen/bootstrap.min.css';
+import 'font-awesome/css/font-awesome.css'
 ReactDOM.render(
   <React.StrictMode>
     <App />
